pkg/codegen/templates/test-client.tmpl

```diff
@@ -65,6 +73,21 @@ func (c *{{$op.OperationId}}TestResponse) Respond{{.ResponseName}}(code int) {{$
     c.tc.parseJSONResponse(c.t, c.Response, &resp)
     return resp
 }
+{{- else if or (eq .ResponseName "4XX") (eq .ResponseName "5XX") }}
+// Error{{.ResponseName}} asserts an error response with the given API error code
+func (c *{{$op.OperationId}}TestResponse) Error{{.ResponseName}}(code APIErrorCode) {{$respType}} {
+    if c.StatusCode != code.HTTPStatus() {
+        c.t.Fatalf("Expected status code %d, got %d", code.HTTPStatus(), code)
+    }
+    var resp {{$respType}}
+    c.tc.parseJSONResponse(c.t, c.Response, &resp)
+
+    if resp.Code != code.AppCode() {
```

Comment:
Note: This implies that there is a `.Code` field in the `{{$respType}}`. Which I think is fine, because we're getting compilation errors at worst, and we really only use one structure for errors (for now).